### PR TITLE
Use keypair version to generate secret key in massa client

### PR DESCRIPTION
**Pull request by sydhds**
* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
* [ ] unit tests on the added/changed features
  * [ ] make tests compile
  * [ ] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

### CODE REVIEW

File: massa-client/src/cmds.rs [hunk 1]
```diff
@@ -782,27 +782,37 @@ impl Command {
...
+
+                            if is_about_keypair && state == grpc::ComponentStateId::Active {
+                                Some(entry.state_id)
+                            } else {
+                                None
                             }
                         })
                         .unwrap_or(0);
-                    println!("Should create address with version: {}", _address_version);
+                    let key = KeyPair::generate(keypair_version as u64).unwrap();
```
[the remark this answers is not in the document]
Comment (sydhds) [reply massa-client/src/cmds.rs:814]: If the version is unknown, there will be an error message like:
* "KeyPair version {} doesn't exist."

Looks good to me no?

[the remark this answers is not in the document]
Comment (Leo-Besancon) [reply massa-client/src/cmds.rs:814]: Ok sure 👍

**Review by Leo-Besancon — APPROVED** [review]

File: massa-versioning/src/grpc_mapping.rs [hunk 2]
```diff
@@ -15,13 +15,23 @@ impl From<&ComponentStateTypeId> for grpc::ComponentStateId {
...
+impl From<&MipComponent> for grpc::MipComponent {
+    fn from(value: &MipComponent) -> Self {
+        match value {
+            MipComponent::KeyPair => grpc::MipComponent::Keypair,
+            MipComponent::Address => grpc::MipComponent::Address,
+            _ => grpc::MipComponent::Unspecified,
+        }
+    }
+}
+
```
Comment (aoudiamoncef) [inline massa-versioning/src/grpc_mapping.rs:27]: IMHO, if you are mirroring your Rust struct and  proto ones, just use the identifier without explicite mapping. 

The coherence is assured by the fact that we always must add new enums at the end of the enum. There is a CI in proto files to detect breaking changes

File: massa-proto/proto/massa/api/v1/versioning.proto [hunk 3]
```diff
@@ -46,3 +46,10 @@ enum ComponentStateId {
   COMPONENT_STATE_ID_ACTIVE = 5;
   COMPONENT_STATE_ID_FAILED = 6;
 }
+
+// Same as MipComponent enum in versioning package
```
Comment (aoudiamoncef) [inline massa-proto/proto/massa/api/v1/versioning.proto:50]: In proto files, always copy/paste the doc on Rust struct. This is will be added to autogenerated documentation

File: massa-versioning/src/grpc_mapping.rs (hunk 2 above)
Comment (aoudiamoncef) [reply massa-versioning/src/grpc_mapping.rs:27]: My bad, It's a new case and It's totally different than existing mappings. So, this the way. I have spoken

File: massa-proto/proto/massa/api/v1/versioning.proto (hunk 3 above)
Comment (sydhds) [reply massa-proto/proto/massa/api/v1/versioning.proto]: Updated

**Review by aoudiamoncef — APPROVED** [review]
Good job 👍🏻

**Pull request merged**